File: fetch_news.py
import urllib.request
import urllib.parse
import xml.etree.ElementTree as ET

queries = ['지정학 물류', '스마트 항만', '항만 크레인 미국']

for q in queries:
    url = f"https://news.google.com/rss/search?q={urllib.parse.quote(q)}&hl=ko&gl=KR&ceid=KR:ko"
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    try:
        html = urllib.request.urlopen(req).read()
        root = ET.fromstring(html)
        print(f"\n--- {q} ---")
        for item in root.findall('.//item')[:3]:
            title = item.find('title').text
            link = item.find('link').text
            print(f"{title} | {link}")
    except Exception as e:
        print(f"Error fetching {q}: {e}")

File: fetch_news.js
const https = require('https');

const queries = [
    { title: 'Geopolitics', q: encodeURIComponent('항만 해운 지정학 리스크') },
    { title: 'Physical AI', q: encodeURIComponent('부산항만공사 스마트 항만 AI') },
    { title: 'Abroad', q: encodeURIComponent('한국 항만 크레인 미국 수주') }
];

async function fetchNews() {
    for (const query of queries) {
        const url = `https://news.google.com/rss/search?q=${query.q}&hl=ko&gl=KR&ceid=KR:ko`;
        
        await new Promise((resolve) => {
            https.get(url, (res) => {
                let data = '';
                res.on('data', chunk => data += chunk);
                res.on('end', () => {
                    console.log(`\n--- ${query.title} ---`);
                    const items = data.split('<item>').slice(1, 4); // get first 3 items
                    items.forEach(item => {
                        const titleMatch = item.match(/<title>(.*?)<\/title>/);
                        const linkMatch = item.match(/<link>(.*?)<\/link>/);
                        if (titleMatch && linkMatch) {
                            console.log(`TITLE: ${titleMatch[1]}`);
                            console.log(`LINK: ${linkMatch[1]}`);
                        }
                    });
                    resolve();
                });
            });
        });
    }
}

fetchNews();
